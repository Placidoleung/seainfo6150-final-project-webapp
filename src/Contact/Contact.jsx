import React, { useState } from "react";
import styles from './Contact.module.css';
import Form from "../Form/Form";
import NavBar from "../NavBar/NavBar.jsx";

const Contact = (props) => {
    return (
        <div className={styles.container}>
            <NavBar/>
            <div className={styles.logo}>
                <a>Contact BlockNews</a>
            </div>
            <div className={styles.form}>
                <Form />
            </div>
        </div>
    )
}

export default Contact
