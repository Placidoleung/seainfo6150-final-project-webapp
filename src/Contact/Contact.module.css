@import url('https://fonts.googleapis.com/css2?family=Montserrat&family=Orelega+One&family=Zen+Dots&display=swap');

.logo {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 200px;
    background-color: #111;
}

.logo a{
    color: white;
    text-decoration: none;
    font-family: 'Zen Dots', cursive;
    font-size: xx-large;
}

.form{
    margin-top: 30px;
}