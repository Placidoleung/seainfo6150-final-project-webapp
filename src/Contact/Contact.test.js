import React from 'react'
import Contact from './Contact'

describe("Contact tests", () => {

    it('render correctly ', () => {
        const { container } = render(<Contact />);
        expect(container).toMatchSnapshot();
    });

});