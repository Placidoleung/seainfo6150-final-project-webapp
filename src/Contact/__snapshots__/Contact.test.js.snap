// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact tests render correctly  1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <div
        class="logo"
      >
        <a
          href="/"
        >
          BlockNews
        </a>
      </div>
      <ul
        class="nav"
      >
        <li
          class="item"
        >
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="item"
        >
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
        <li
          class="item"
        >
          <a
            href="/about"
          >
            About
          </a>
        </li>
      </ul>
    </div>
    <div
      class="logo"
    >
      <a>
        Contact BlockNews
      </a>
    </div>
    <div
      class="form"
    >
      <div
        class="container"
      >
        <form
          class="form"
        >
          <label
            for="firstName"
          >
            First Name*
          </label>
          <input
            id="firstName"
            name="firstName"
            required=""
            type="text"
          />
          <label
            for="lastName"
          >
            Last Name*
          </label>
          <input
            id="lastName"
            name="lastName"
            required=""
            type="text"
          />
          <label
            for="email"
          >
            Email*
          </label>
          <input
            id="email"
            name="email"
            required=""
            type="email"
          />
          <label
            for="tel"
          >
            Telephone
          </label>
          <input
            id="tel"
            name="tel"
            pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
            placeholder="xxx-xxx-xxxx"
            type="tel"
          />
          <label
            for="usingTime"
          >
            How long have you been using BlockNews?
          </label>
          <select
            id="usingTime"
            name="usingTime"
          >
            <option
              value="day"
            >
              Less than a week
            </option>
            <option
              value="week"
            >
              Several weeks
            </option>
            <option
              value="month"
            >
              Several months
            </option>
            <option
              value="year"
            >
              More than a year
            </option>
          </select>
          <label
            for="often"
          >
            How often do you use BlockNews? 
          </label>
          <select
            id="often"
            name="often"
          >
            <option
              value="day"
            >
              Everyday
            </option>
            <option
              value="week"
            >
              Every Week
            </option>
            <option
              value="month"
            >
              Every Month
            </option>
            <option
              value="rarely"
            >
              Rarely
            </option>
          </select>
          <label
            for="message"
          >
            Your Message*
          </label>
          <input
            id="message"
            name="message"
            required=""
            type="text"
          />
          <label>
            May we contact you to follow up?
          </label>
          <div
            class="contact"
          >
            <label
              for="contact-yes"
            >
              Yes
            </label>
            <input
              id="contact-yes"
              name="contact"
              type="radio"
              value="yes"
            />
            <label
              for="contact-no"
            >
              No
            </label>
            <input
              id="contact-no"
              name="contact"
              type="radio"
              value="no"
            />
          </div>
          <input
            type="submit"
            value="Send"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;
